File: scala/src/main/scala/TwoSum.scala
package leetcode

object TwoSum {
  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    var map:Map[Int, Int] = Map()
    for (i <- 0 to nums.length) {
      if (map.contains(target - nums(i))) return Array(map(target - nums(i)), i)
      else map += (nums(i) -> i)
    }
    Array(0, 0)
  }
}

File: python/src/two_sum.py
from typing import List

class TwoSum:
    def solution(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i, v in enumerate(nums):
            if target - nums[i] in map:
                return [map[target - nums[i]], i]
            else:
                map[nums[i]] = i
